### PR TITLE
Column Selection DSL improvements

### PR DESCRIPTION
**Pull request by Jolanrensen**
Based on https://github.com/Kotlin/dataframe/issues/317
I went over the Column Selection DSL and added support for KProperties everywhere.
Additionally, I saw there were some missing APIs for Strings as well, like
`select { "myGroup".allAfter("someCol") }`, `select { "myCol".groups { true } }` etc.

Finally, I renamed `allUntil` to `allUpTo`, as "until" means excluding the last one, while "up to" does not. And I renamed `allSince` to `allFrom`.

Docs are updated too.

### CODE REVIEW

File: core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/samples/api/Access.kt [hunk 1]
```diff
@@ -706,6 +706,41 @@ class Access : TestBase() {
...
+        df.select { Person::name[Name::firstName].map { it.lowercase() } }
+
+        // column arithmetics
+        df.select { 2021 - (Person::age)() }
+
+        // two columns
+        df.select { Person::name and Person::age }
+
+        // range of columns
+        df.select { Person::name..Person::age }
```
Comment (zaleslaw) [inline core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/samples/api/Access.kt:734]: Very strange API:)

Comment (Jolanrensen) [reply core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/samples/api/Access.kt:734]: not weirder than "name".."age" imo

Comment (zaleslaw) [reply core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/samples/api/Access.kt:734]: Of course not, I mean totally the range of columns based on name, I could imagine ranging the ordered things or alphabetically, but not the range of column names in the structure, is this order guaranteed by compiler or somehow?

Comment (Jolanrensen) [reply core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/samples/api/Access.kt:734]: Nope, that's impossible afaik. However, it's still very useful to select multiple columns quickly. But, as with all non-typesafe operations (resulting in `Any?` types), it works best in Jupyter or with the upcoming plugin, which generate accessors on the fly.

**Comment by Jolanrensen** [comment]
If we don't merge this PR, it will be joint with https://github.com/Kotlin/dataframe/pull/331, since that builds on top of this PR adding docs and other missing functions

**Pull request merged**